### ISSUE TITLE
Hyper-params to train dfl method on portfolio problem

### ISSUE THREAD

**Issue by dishank-b** [sanketkshah/LODLs#1]
Can you provide the hyper-params you used to train dfl on the portfolio problem such as `lr`, `batchsize`, etc? We are not able to reproduce the numbers from the paper, of dfl on portfolio problem, so if you can provide with the exact hyper-params to reproduce the number from the paper. Though, the MSE on portfolio matches to what's in the paper. Thanks.

**Comment by sanketkshah** [comment]
Unfortunately, I didn't save the hyperparameters for the experiments at the time, so I don't have the exact values. However, one thing that I remember as being important is the learning rate; I think 0.01 worked best, but maybe it was 0.1? Also, I used the largest batch size that my memory would allow.

If reproducing dfl is an issue, however, the reason may be that the `dflalpha` hyperparameter needs to be tuned. This parameter controls the amount of 'two-stage' loss that's added to the dfl loss as regularization. In my experiments, I think I tried the values `0.01`, `0.1`, `1`, and `10`; the rule of thumb that I followed was to set `dflalpha` such that the dfl loss and the two-stage loss are of the same order of magnitude.

Let me know if you still can't reproduce the results, though... Happy to get on a call to chat about this!

**Comment by dishank-b** [comment]
Thanks. Using `dflalpha=10`, got closer to the reported results. 
On similar note, what's the actual value you for `--numsamples` param, in README for two problems you say it being `50000` but on the other it's just `5000`. In the paper, you seems to go up until `5000` only. Can you tell what's the correct value for `--numsamples` to use to get the reported numbers in table 1.

**Comment by dishank-b** [comment]
Also, if you have the hyper-parameters noted down to reproduce the results from the paper in table 1. That would be awesome. Thanks

**Comment by sanketkshah** [comment]
For Table 1, I used 5000 samples. I used 50000 in the readme because that more consistently gives good results, but this can be a bit of an overkill.

As for the remaining hyperparams, unfortunately, I didn't save the hyperparameters for the experiments at the time, so I don't have the exact values...

**Comment by dishank-b** [comment]
Do you remember anything specific about training using `quad` method for `portfolio` problem? I am getting results very far from what reported in the paper. I am using following hyper params:
`--problem=portfolio --loss=quad --seed=$i --directory=$dir --instances=400 --testinstances=400 --valfrac=0.5 --stocks=50 --stockalpha=0.1 --lr=0.01 --sampling=random --samplingstd=0.1 --numsamples=5000 --losslr=0.001 --serial=False --batchsize=100` and default params from the code of `quadrank=20` and `quadalpha=0`.
